packages/@react-spectrum/s2/src/SegmentedControl.tsx

```diff
@@ -156,16 +189,97 @@ function DefaultSelectionTracker(props: DefaultSelectionTrackProps) {
     }
   }, [])
 
+  // First: get the current bounds of the element
+  useEffect(() => {
+    if (domRef.current) {
+      let item = domRef.current.querySelector("label[data-selected=true]");
+      itemRef.current = item;
+
+      // i don't really want to have this here, and honestly, it should probably go in useLayoutEffect
+      // however, then i basically have to pull all the code currently in useEffect to useLayoutEffect which feels repetitive so I'm just keeping it here for now
+      let styleObj: { transform: string | undefined, width: string | undefined, height: string | undefined } = {
+        transform: undefined,
+        width: undefined,
+        height: undefined
+      };
+
+      let finalW = item?.offsetWidth;
+      let finalH = item?.offsetHeight;
+
+      styleObj.width = finalW;
+      styleObj.height = finalH;
+      setStyle(styleObj);
+
+    }
+  }, [state?.selectedValue])
+
+  // Last: get the final bounds of the element
+  useLayoutEffect(() => {
+    if (domRef.current) {
+      let slide = document.querySelector('#animate');
+      let item = domRef.current.querySelector("label[data-selected=true]")
+      let cachedItem = itemRef?.current?.getBoundingClientRect();
+
+
+      if (itemRef.current) {
+        let finalItem = item.getBoundingClientRect();
+
+        // Invert: determine the delta between the first and last bounds of the element
+        // let deltaX = cachedItem.left - finalItem.left;
+        // let deltaW = cachedItem.width / finalItem.width;
+        // let deltaW = finalItem.width / cachedItem.width;
+        // let deltaH = cachedItem.height / finalItem.height;
+        let cachedX = itemRef?.current.offsetLeft;
+        let finalX = item.offsetLeft;
+        // let finalW = item.offsetWidth;
+        // let finalH = item.offsetHeight;
+
+        // styleObj.width = finalW;
+        // styleObj.height = finalH;
+        // setStyle(styleObj);
+        
+        // Play: animate the final element from its first bounds to its last bound
+        slide.animate(
+          [
+            {transform: `translateX(${finalX}px)` },
+          ],
+          {
+            duration: 150,
```

Comment:
We may want to consider having the duration to be dynamic based on the deltaX so that it always goes the same speed.

Reply:
might bring up to design and ask them what they prefer

Reply:
Thanks for providing both. For those testing, it's easiest to see the difference when switching between the left-most and rightmost controls. I like the second *better*, but it does still seem a little slow. I'm curious what design's feedback is. Maybe a middle ground (?)

Reply:
Seems like design (or more like Lindsey since she was the one I shared this with) liked the first one better. Her reasoning was that the second one felt like it was taking a long time between when you clicked on end to the other end. She also worried that it would feel even more slow if there were more items between the two.  

That said, if it feels too slow, we can adjust the speed to be quicker. However, that would also mean for shorter distances, the animation would be pretty fast.